src/main/java/com/hierynomus/smbj/share/File.java

```diff
@@ -114,14 +115,22 @@ public int read(byte[] buffer, long fileOffset) throws IOException {
      * @param fileOffset The offset, in bytes, into the file from which the data should be read
      * @param offset the start offset in the buffer at which to write data
      * @param length the maximum number of bytes to read
-     * @return the actual number of bytes that were read
+     * @return the actual number of bytes that were read; or -1 if the end of the file was reached
      */
     public int read(byte[] buffer, long fileOffset, int offset, int length) throws IOException {
         SMB2ReadResponse response = share.read(fileId, fileOffset, length);
-        byte[] data = response.getData();
-        int bytesRead = Math.min(length, data.length);
-        System.arraycopy(data, 0, buffer, offset, bytesRead);
-        return bytesRead;
+        if (response.getHeader().getStatus() == NtStatus.STATUS_END_OF_FILE) {
+            return -1;
```

Comment:
Would it be ignoring the actual number of bytes that were already read into the buffer before EOF if -1 is returned here?